packages/contracts-bedrock/test/L1/StandardValidator.t.sol

```diff
@@ -0,0 +1,890 @@
+import { IProxyAdmin } from "interfaces/universal/IProxyAdmin.sol";
+import { IDelayedWETH } from "interfaces/dispute/IDelayedWETH.sol";
+import { IPreimageOracle } from "interfaces/cannon/IPreimageOracle.sol";
+import { ISemver } from "interfaces/universal/ISemver.sol";
+import { IResourceMetering } from "interfaces/L1/IResourceMetering.sol";
+import { IOptimismPortal2 } from "interfaces/L1/IOptimismPortal2.sol";
+import { IDisputeGame } from "interfaces/dispute/IDisputeGame.sol";
+import { IMIPS } from "interfaces/cannon/IMIPS.sol";
+
+contract StandardValidatorV180Testing is StandardValidatorV180 {
+    mapping(address => bytes32) internal validHashes;
+
+    constructor(
+        ISuperchainConfig _superchainConfig,
+        address _l1PAOMultisig,
+        address _mips
+    )
+        StandardValidatorV180(_superchainConfig, _l1PAOMultisig, _mips)
+    { }
+
+    function bytecodeHashEq(address _addr, bytes32 _hash) internal view override returns (bool) {
+        return validHashes[_addr] == _hash;
+    }
+
+    function setValidHash(address _addr, bytes32 _hash) external {
+        validHashes[_addr] = _hash;
+    }
+}
+
+contract StandardValidatorV180_Test is Test {
+    StandardValidatorV180Testing validator;
+    ISuperchainConfig superchainConfig;
+    address l1PAOMultisig;
+    address mips;
+
+    // Mock contracts for validation
+    IProxyAdmin proxyAdmin;
+    ISystemConfig systemConfig;
+    bytes32 absolutePrestate;
+    uint256 l2ChainID;
+
+    // Mock addresses for dependencies
+    address disputeGameFactory;
+    address permissionedDisputeGame;
+    address anchorStateRegistry;
+    address optimismPortal;
+    address l1CrossDomainMessenger;
+    address l1StandardBridge;
+    address l1ERC721Bridge;
+    address optimismMintableERC20Factory;
+    address delayedWETH;
+    address preimageOracle;
+
+    function setUp() public {
+        // Setup test addresses
+        l1PAOMultisig = makeAddr("l1PAOMultisig");
+        mips = makeAddr("mips");
+        address guardian = makeAddr("guardian");
+        superchainConfig = ISuperchainConfig(makeAddr("superchainConfig"));
+
+        // Mock superchainConfig calls needed in setup
+        vm.mockCall(address(superchainConfig), abi.encodeCall(ISuperchainConfig.guardian, ()), abi.encode(guardian));
+        vm.mockCall(address(superchainConfig), abi.encodeCall(ISuperchainConfig.paused, ()), abi.encode(false));
+
+        // Deploy validator
+        validator = new StandardValidatorV180Testing(superchainConfig, l1PAOMultisig, mips);
+
+        // Setup mock contracts for validation
+        proxyAdmin = IProxyAdmin(makeAddr("proxyAdmin"));
+        systemConfig = ISystemConfig(makeAddr("systemConfig"));
+        absolutePrestate = bytes32(uint256(0xdead));
+        l2ChainID = 10;
+
+        // Setup mock dependency addresses
+        disputeGameFactory = makeAddr("disputeGameFactory");
```

Comment:
Seems stronger to deploy a chain with OPCM here to make sure everything is realistic, and then we make sure everything passes for standard OPCM chains?